db/tx.go

```diff
@@ -100,12 +134,17 @@ func (DBI *DB) FetchTxType() (txType uint64, err error) {
 func getWitnessTranfer(bz bazooka.Bazooka, DBI DB, tx core.Tx) (fromMerkleProof, toMerkleProof bazooka.StateMerkleProof, txDBConn DB, err error) {
 	dbCopy, _ := NewDB(bz.Cfg)
 
+	from, to, err := bz.FetchFromAndToStateIDs(tx)
+	if err != nil {
+		return
+	}
+
 	// fetch from state MP
-	err = DBI.fetchMPWithID(tx.From, &fromMerkleProof)
+	err = DBI.fetchMPWithID(from, &fromMerkleProof)
 	if err != nil {
 		return
 	}
-	toState, err := DBI.GetStateByIndex(tx.To)
+	toState, err := DBI.GetStateByIndex(to)
```

Comment:
So if the tx is a mass migration. `FetchFromAndToStateIDs` returns `to` stateID as 0, and we are fetching stateID 0 for no reason?

Reply:
yeah good eye, created much simpler mass migrations witness fetcher.